dockstore-common/src/main/java/io/dockstore/common/DescriptorLanguageSubclass.java

```diff
@@ -83,4 +84,8 @@ public static DescriptorLanguageSubclass convertShortNameStringToEnum(final Stri
             .filter(subclass -> subclass.getShortName().equalsIgnoreCase(descriptorSubclass)).findFirst();
```

Comment:
Ditto here, seems like we would know the mappings at start-up. Document both methods if misunderstood (or maybe even if understood)

Reply:
This method converts an arbitrary string to the enum, not possible to precompute for all possible strings.

Reply:
Again, I'm not sure if I understand what is going on here, hence the request for documentation. 

But as a strawman for understanding, isn't this just a map?  As in, the set of possible strings is arbitrary, but the set of valid strings is not.  Like `Map<String, DescriptorLanguageSubclass>` with the short name as a key (putting aside case-sensitivity)

Reply:
Was pondering this, one reusable implementation of this would be a `CacheLoader` in Guava
https://www.baeldung.com/guava-cacheloader